stubs/tensorflow/tensorflow/python/feature_column/feature_column_v2.pyi

```diff
@@ -0,0 +1,271 @@
+# The types here are all undocumented, but all feature columns are return types of the
+# public functions in tf.feature_column. As they are undocumented internals while some
+# common methods are included, they are incomplete and do not have getattr Incomplete fallback.
+from _typeshed import Incomplete
+from abc import ABC, ABCMeta, abstractmethod
+from collections.abc import Callable, Sequence
+from typing_extensions import Literal, Self, TypeAlias
+
+import tensorflow as tf
+from tensorflow import _ShapeLike
+
+_Combiners: TypeAlias = Literal["mean", "sqrtn", "sum"]
+_ExampleSpec: TypeAlias = dict[str, tf.io.FixedLenFeature | tf.io.VarLenFeature]
+
+class FeatureColumn(ABC):
+    @property
+    @abstractmethod
+    def name(self) -> str: ...
+    @property
+    @abstractmethod
+    def parse_example_spec(self) -> _ExampleSpec: ...
+    def __lt__(self, other: FeatureColumn) -> bool: ...
+    def __gt__(self, other: FeatureColumn) -> bool: ...
+    @property
+    @abstractmethod
+    def parents(self) -> list[FeatureColumn | str]: ...
+
+class DenseColumn(FeatureColumn, metaclass=ABCMeta): ...
+class SequenceDenseColumn(FeatureColumn, metaclass=ABCMeta): ...
+
+# These classes are mostly subclasses of collections.namedtuple but we can't use
+# typing.NamedTuple because they use multiple inheritance with other non namedtuple classes.
+# _cls instead of cls is because collections.namedtuple uses _cls for __new__.
+class NumericColumn(DenseColumn):
+    key: str
```

Comment:
As these are attributes from collections.namedtuple subclassing they are read only and could maybe be marked property? Sorta neutral to being more precise here unsure namedtuple fields are properties either. Especially as right now it's little more lenient and all these classes aren't documented so being stricter seems unnecessary.